modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/httpclient/EntityStatementJwtConverter.kt

```diff
@@ -19,27 +19,20 @@ class EntityStatementJwtConverter: ContentConverter {
         charset: Charset,
         typeInfo: TypeInfo,
         value: Any?
-    ): OutgoingContent? {
-        if (value is EntityStatement) {
-            return OutgoingEntityStatementContent(value)
-        } else if (value is String) {
-            JsonMapper().mapEntityStatement(value)?.let {
-                return OutgoingEntityStatementContent(it)
-            }
-        }
-        return null
+    ): OutgoingContent? = when (value) {
+        is EntityStatement -> OutgoingEntityStatementContent(value)
+        is String -> JsonMapper().mapEntityStatement(value)?.let { OutgoingEntityStatementContent(it) }
+        else -> null
     }
 
-    override suspend fun deserialize(charset: Charset, typeInfo: TypeInfo, content: ByteReadChannel): Any? {
-        val text = content.readRemaining().readText(charset)
-        return Json.decodeFromString(EntityStatement.serializer(), text)
+    override suspend fun deserialize(charset: Charset, typeInfo: TypeInfo, content: ByteReadChannel): Any =
+        content.readRemaining().readText(charset).let {
+            Json.decodeFromString(EntityStatement.serializer(), it)
     }
 }
 
 class OutgoingEntityStatementContent(private val entityStatement: EntityStatement): OutgoingContent.ByteArrayContent() {
 
-    override fun bytes(): ByteArray {
-        val serializedData = Json.encodeToString(entityStatement)
-        return serializedData.toByteArray(Charsets.UTF_8)
-    }
+    override fun bytes(): ByteArray =
+        Json.encodeToString(entityStatement).toByteArray(Charsets.UTF_8)
```

Comment:
What is the intent here? Where are these bytes needed? A JSON directly to URF-8 bytes is not used very often. Please point me to the relevant requirement in the spec or in a ticket

Reply:
That's actually the default:

<img width="771" alt="Scherm­afbeelding 2024-07-30 om 16 05 36" src="https://github.com/user-attachments/assets/d4a7a366-165d-40eb-91ee-ba1fef4194ac">

Reply:
This is part of the content negotiation of Ktor. The interface requires the content to be a ByteArray: https://ktor.io/docs/serialization.html#implement_custom_serializer

Reply:
Here is the part of the spec that I'm implementing: https://openid.net/specs/openid-federation-1_0.html#name-fetch-entity-statement-resp

Reply:
So these should be jwt strings...

Reply:
Refactored to return jwt string